ui/src/pages/AllStories.tsx

```diff
@@ -0,0 +1,288 @@
+import Paper from "@material-ui/core/Paper";
+import {
+  createStyles,
+  Theme,
+  WithStyles,
+  withStyles,
+} from "@material-ui/core/styles";
+import Switch from "@material-ui/core/Switch";
+import TableCell from "@material-ui/core/TableCell";
+import clsx from "clsx";
+import * as React from "react";
+import {
+  AutoSizer,
+  Column,
+  Table,
+  TableCellRenderer,
+  TableHeaderProps,
+} from "react-virtualized";
+import useSWR from "swr";
+
+import { colors } from "../styles/colors";
+import { Story } from "../types";
+
+declare module "@material-ui/core/styles/withStyles" {
+  interface BaseCSSProperties {
+    flip?: boolean;
+  }
+}
+
+const styles = (theme: Theme) =>
+  createStyles({
+    flexContainer: {
+      display: "flex",
+      alignItems: "left",
+      boxSizing: "border-box",
+    },
+    table: {
+      "& .ReactVirtualized__Table__headerRow": {
+        flip: false,
+        paddingRight: theme.direction === "rtl" ? "0 !important" : undefined,
+        backgroundColor: colors.primaryLight4,
+        fontWeight: 800,
+      },
+      "& .ReactVirtualized__Grid ReactVirtualized__Table__Grid": {},
+    },
+    tableRow: {
+      cursor: "pointer",
+      color: colors.grey,
+    },
+    tableRowHover: {
+      "&:hover": {
+        backgroundColor: colors.black,
+      },
+    },
+    tableCell: {
+      flex: 2,
+    },
+    noClick: {
+      cursor: "initial",
+    },
+  });
+
+interface ColumnData {
+  dataKey: string;
+  label: string;
+  numeric?: boolean;
+  toggle?: true;
+  width: number;
+}
+
+interface Row {
+  index: number;
+}
+
+interface MuiVirtualizedTableProps extends WithStyles<typeof styles> {
+  columns: ColumnData[];
+  headerHeight?: number;
+  onRowClick?: () => void;
+  rowCount: number;
+  rowGetter: (row: Row) => TableData;
+  rowHeight?: number;
+}
+
+class MuiVirtualizedTable extends React.PureComponent<
```

Comment:
Can we extract this into it's own component (`Table.tsx`) and make them all functional? We aren't using any of the class based lifecycles.

We can just turn the class function into standalone functions.

Reply:
it costs nothing to pass down additional function args instead of using `this.props`